content/200-orm/800-more/600-help-and-troubleshooting/100-help-articles/430-nextjs-prisma-client-dynamic.mdx

```diff
@@ -0,0 +1,30 @@
+---
+title: 'Best practices for using Prisma Client with Next.js and Vercel caching'
+metaTitle: 'Best practices for using Prisma Client with Next.js and Vercel caching'
+metaDescription: 'Learn best practices to avoid issues with route caching in Next.js and Vercel'
+---
+
+## Problem
+
+When deploying a Next.js app to Vercel, you may run into an issue where your queries are not updating or displaying the right content. Or, you may find that it takes a long time for newly created objects to show up in your queries.
+
+In these cases, you are most likely running into a feature of Next.js: [automatic caching of routes](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic)
+
+Since routes are cached in many cases, the results from any queries are also cached, meaning that any updates to your data would not be reflected in your page until the cache expires.
+
+## Solution
+
+In these cases, it's important to [set the dynamic behavior of the route](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic) to `"force-dynamic"` so that up to date data is always returned.
```

Comment:
```suggestion
In these cases, it's important to [set the dynamic behavior of the route](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic) to `"force-dynamic"` so that up to date data is always returned and no caching happens.
```